mqtt_bridge/bridge.py

```diff
@@ -37,63 +38,63 @@ class Bridge(object, metaclass=ABCMeta):
 
 class RosToMqttBridge(Bridge):
     """ Bridge from ROS topic to MQTT
-
     bridge ROS messages on `topic_from` to MQTT topic `topic_to`. expect `msg_type` ROS message type.
     """
 
-    def __init__(self, topic_from: str, topic_to: str, msg_type: rospy.Message, frequency: Optional[float] = None):
+    def __init__(self, topic_from: str, topic_to: str, msg_type, frequency: Optional[float] = None, **kwargs):
+        self.ros_node = kwargs["ros_node"]
         self._topic_from = topic_from
         self._topic_to = self._extract_private_path(topic_to)
-        self._last_published = rospy.get_time()
-        self._interval = 0 if frequency is None else 1.0 / frequency
-        rospy.Subscriber(topic_from, msg_type, self._callback_ros)
+        self._last_published = self.ros_node.get_clock().now()
+        self._interval = Duration(seconds=0) if frequency is None else Duration(seconds=(1.0 / frequency))
+        self.ros_node.create_subscription(msg_type, topic_from, self._callback_ros, 10)
 
-    def _callback_ros(self, msg: rospy.Message):
-        rospy.logdebug("ROS received from {}".format(self._topic_from))
-        now = rospy.get_time()
+    def _callback_ros(self, msg):
```

Comment:
Can you add type hinting to `msg` argument? I'm not sure what it is in ROS2.

Reply:
ROS2:
<img width="284" alt="image" src="https://user-images.githubusercontent.com/36097171/165628154-30818e45-014e-40c6-b986-d3fbd59e8d9d.png">

ROS1:
<img width="644" alt="image" src="https://user-images.githubusercontent.com/36097171/165628297-6c960d94-70af-4341-b9c2-b0566333fc39.png">

Again the same thing ros2 library does not contain a generic class like AnyMsg or Message. So  type hinting is not possible right now.

Reply:
I see, thanks.